### PR TITLE
Adding project for Site Recovery CSMv2 based specs.

### PR DESCRIPTION
**Pull request by NehaRaje**
Adding a new project for specs of Site Recovery Services based on the CSMv2 APIs.
API functionality has been added only for the enterprise to enterprise replication provider.
Basic framework for tests has been provided and test added.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@NehaRaje__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@NehaRaje__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

**Comment by stankovski** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/SiteRecovery/SiteRecoveryManagement/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+using System.Reflection;
+
+[assembly: AssemblyTitle("Microsoft.Azure.Management.RecoveryServicesARM.Specification")]
```
Comment (stankovski) [inline src/ResourceManagement/SiteRecovery/SiteRecoveryManagement/Properties/AssemblyInfo.cs]: Add AssemblyVersion and AssemblyFileVersion explicitly

File: src/ResourceManagement/SiteRecovery/SiteRecoveryManagement/Microsoft.Azure.Management.RecoveryServicesARM.nuget.proj [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup>
+    <!--
+    Microsoft.Azure.Management.RecoveryServicesARM
+    -->
+    <SdkNuGetPackage Include="Microsoft.Azure.Management.RecoveryServicesARM">
```
Comment (hovsepm) [inline src/ResourceManagement/SiteRecovery/SiteRecoveryManagement/Microsoft.Azure.Management.RecoveryServicesARM.nuget.proj]: please remove ARM suffix from the package name in the nuget.proj and nuspec files and their names.

**Pull request merged**